### ISSUE TITLE
Automate windows CLI testing

### ISSUE THREAD

**Issue by grampelberg** [linkerd/linkerd2#2055]
### What problem are you trying to solve?

When adding new CLI functionality, there are platform differences that get in the way of features working (such as #2014). It is frustrating to have end users find these as part of using a stable milestone.

### How should the problem be solved?

Add *very* basic windows testing to the CI runs. This is to solve something obviously not working instead of in depth integration or unit testing on windows.

**Comment by jon-walton** [comment]
something to keep in mind is that from an application perspective, Windows != WSL. WSL = "Linux" (usually Ubuntu userspace with a "kernel").

I've personally not experienced any bugs running `linkerd.exe` on windows but can reproduce some of the issues mentioned for WSL depending on the version of windows (remember how I said windows != WSL? ha!)

Most of the issues (with other apps too) tend to be 1 of 2 things:

1. WSL isn't running a window manager so it gets tricky if you want to start, lets say, firefox. You need to reach into windows and start firefox.exe (WSL exposes the windows `$path` which let's that happen, the exe will run in windows with stdout / stderr piped back into WSL. or the full path `$ /mnt/c/Program\ Files\ \(x86\)/Mozilla\ Firefox/firefox.exe`)

2. The linux version of the cli is trying to use kernel features that haven't been implemented yet. This is where you'll get into some weird compatibility matrix where the linux build of linkerd cli doesn't work in older versions of windows, but does in newer ones. The WSL kernel is tied to the version of windows (it's a clean room implementation and not the actual linux kernel)

Not sure if that's clear, I've not had breakfast yet 😄 

tl;dr

You'll want to test both

Windows - Windows (`GOOS=windows`)
Windows - WSL (`GOOS=linux`)

**Comment by grampelberg** [comment]
@jon-walton that's great clarity!

**Comment by grampelberg** [comment]
We are getting this in a very small degree from the CI test in linkerd/website now.
